BankSDK/GiniBankSDK/Sources/GiniBankSDK/Extensions/Foundation/Date+Formatting.swift

```diff
@@ -26,4 +26,31 @@ extension Date {
         let berlinDate = calendar.date(byAdding: .second, value: calendar.timeZone.secondsFromGMT(for: self), to: self)
         return berlinDate ?? Date()
     }
+    
+    func isDueSoon(within days: Int) -> Bool {
+        let now = Date()
+        
+        /// If this date is in the past, it's not considered "due soon"
+        guard self >= now else { return false }
+        
+        /// Calculate the latest date that counts as "due soon"
+        guard let upperLimit = Calendar.current.date(byAdding: .day, value: days, to: now) else { return false }
+        
+        /// Return true if this date falls within the upcoming 'days' range
+        return self <= upperLimit
+    }
+    
+    func toDisplayString(format: String = "dd.MM.yyyy") -> String {
+        let formatter = DateFormatter()
+        formatter.dateFormat = format
+        formatter.locale = Locale.current
+        return formatter.string(from: self)
+    }
+    
+    static func date(fromServerString serverDateString: String) -> Date? {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd"
+        formatter.locale = Locale(identifier: "en_US_POSIX")
```

Comment:
[Suggestion]: we will need test for it to track case when the format has changed